### PR TITLE
eth: update contract bindings and code cleanup

### LINKED ISSUE #652: pm: generate Go contract bindings for LivepeerETHTicketBroker contract
**Is your feature request related to a problem? Please describe.**

The `pm` package includes a Broker interface that represents an object capable of processing deposits and ticket redemptions. In practice, the main implementation of this interface will be a Go contract binding for [LivepeerETHTicketBroker](https://github.com/livepeer/protocol/blob/pm/contracts/pm/LivepeerETHTicketBroker.sol). We must generate this Go contract binding.

**Describe the solution you'd like**

We can generate the contract binding using the `abigen` tool that is a part of the go-ethereum toolchain. We already use `abigen` to generate the rest of our contract bindings.

However, we should investigate the ability of `abigen` to generate a contract binding for a contract that is uses the experimental `ABIEncoderV2` (i.e. LivepeerETHTicketBroker). I made a very quick attempt at using `abigen` to generate the binding for LivepeerETHTicketBroker, but encountered the following error:

```
Failed to generate ABI binding: unsupported arg type: tuple
client.go:14: running "abigen": exit status 255
```

We should investigate this error and if `abigen`, does not currently support contracts that use the experimental `ABIEncoderV2`, we should file a feature request and maybe consider implementing the feature ourselves in order to generate our required contract binding.

**Describe alternatives you've considered**

We can choose to not use the experimental `ABIEncoderV2` in LivepeerETHTicketBroker, but this would force us to list out all parameters in any function signature where we want to provide a struct which hurts code readability and the usability of the contract API.

### PR DESCRIPTION
**Pull request by yondonfu**
**What does this pull request do? Explain your changes. (required)**

Update contract bindings to reflect the latest changes in the protocol contracts [pm](https://github.com/livepeer/protocol/tree/pm) branch. In preparation for integrating the `pm` package with the rest of the go-livepeer codebase and for interactions with the updated contracts, this PR seeks to remove all remaining V1 protocol job related code.

**Specific updates (required)**

- Introduces LivepeerETHTicketBroker contract binding
- Updates the rest of the contract bindings
- Remove job related code

**How did you test each of these updates (required)**

I built the node and ran it in broadcaster/orchestrator mode in devenv

**Does this pull request close any open issues?**

Fixes #652 

**Checklist:**

- [x] README and other documentation updated
- [x] Node runs in OSX and devenv
- [x] All tests in `./test.sh` pass

### CODE REVIEW

File: server/mediaserver.go
Comment (eladmallel) [inline server/mediaserver.go:281]: Perhaps worth capturing as an issue? - I'm concerned we might miss this. wdyt?

**Review by eladmallel — APPROVED** [review]
Besides the one little comment I added, LGTM. However, I'm probably not the best person to review this kind of change.. wondering if perhaps another set of eyes should review this PR?

**Comment by yondonfu** [comment]
Since master is known to be unstable and in the interest of avoiding further merge conflicts with other people working off of master I'd be ok with getting this merged sooner rather than later. I think it is also worth noting that master as is also won't work with the `pm` branch of the protocol repo.

Putting the merge on hold because I'm going to push an update to this PR to make sure LP client adheres to the `pm/Broker` interface

**Comment by yondonfu** [comment]
661f2c9: client implements Broker

Tested by adding the following line in `client.go` which should raise a compiler error if the client did not implement Broker:

```
var _ pm.Broker = client{}
```

File: cmd/livepeer_cli/wizard_stats.go
Comment (j0sh) [inline cmd/livepeer_cli/wizard_stats.go:329]: Don't we still need a deposit?

Comment (yondonfu) [reply cmd/livepeer_cli/wizard_stats.go:329]: Yeah, but it is with a different contract so this PR cleaned up the old code and the new code for the new contract is included in #658 which fetches all of a PM sender's information in a single request instead of individually fetching the sender's deposit and penalty escrow.

**Pull request merged**
